lib/protractor.js

```diff
@@ -720,22 +714,49 @@ Protractor.prototype.initDebugger_ = function(debuggerClientPath, opt_debugPort)
     },
 
     // Execute a piece of code.
+    // Result is a string representation of the evaluation.
     execute: function(code) {
       var execFn_ = function() {
+        var deferred = webdriver.promise.defer();
```

Comment:
Instead of having this deferred, you can simply this by using:

```js
  var res = vm_.runInThisContext(code);
  if(!webdriver.promise.isPromise(res)) {
    res = webdriver.promise.fulfilled(res);
  }
  return res.then(...
```